homeassistant/components/homeassistant/__init__.py

```diff
@@ -275,4 +277,50 @@ async def async_handle_reload_config_entry(call: ha.ServiceCall) -> None:
         schema=SCHEMA_RELOAD_CONFIG_ENTRY,
     )
 
+    async def async_handle_reload_all(call: ha.ServiceCall) -> None:
+        """Service handler for calling all integration reload services.
+
+        Calls all reload services on all active domains, which triggers the
+        reload of YAML configurations for the domain that support it.
+
+        Additionally, it also calls the `homeasssitant.reload_core_config`
+        service, as that reloads the core YAML configuration, and the
+        `frontend.reload_themes` service, as that reloads the themes.
+
+        We only do so, if there are no configuration errors.
+        """
+
+        if errors := await conf_util.async_check_ha_config_file(hass):
+            _LOGGER.error(
```

Comment:
Should we really both log and raise?

Reply:
That is what we do at restart as well, so kept it the same as with that. I'm happy to drop the logging. Shall I?

Reply:
I think we can drop the log here if it's logged by the websocket handler. I think it is:

https://github.com/home-assistant/core/blob/03b47f1931dc2902a54778b118e50725fd587ee1/homeassistant/components/websocket_api/commands.py#L214

Reply:
Although I guess a something else could call the service too.

Reply:
Yup, it could be node-red 🤷

Reply:
We can keep it for now. It's a separate topic.